src/Symfony/Component/Security/Guard/Provider/GuardAuthenticationProvider.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Symfony\Component\Security\Guard\Provider;
```

Comment:
Finally updating this PR :). I don't understand what you're proposing here - when you suggest `Guard\Authentication...` - do you actually mean that, or actually `Symfony\Component\Security\Guard\Authentication...`? I'm not convinced I have the classes organized in the most efficient way, and it's an easy change to still make.